### PR TITLE
Feature/nested attributes

### PR DESCRIPTION
**Pull request by jacquesporveau**
https://www.pivotaltracker.com/story/show/142421627

Update: Moving on to another project today. I will keep an eye for requested changes and perhaps make them on a lunch break or something but this one may sit a while.

### CODE REVIEW

File: app/controllers/admin/people_controller.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+      if Person.update(person_params.keys, person_params.values)
+        flash[:success] = 'People successfully updated.'
+      end
+      redirect_to admin_people_path
+    end
+
+    private
+
+    def person_params
+      params.require(:people).permit(person: [role_ids: []])[:person]
```
Comment (graygilmore) [inline app/controllers/admin/people_controller.rb:17]: 💯 🔥

File: spec/controllers/admin_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
+require 'rails_helper'
+
+RSpec.describe AdminController, type: :controller do
+  describe '#authenticate_admin' do
```
Comment (graygilmore) [inline spec/controllers/admin_controller_spec.rb]: I believe that the controllers that are actually executing this path should cover these tests already. @brendandeere what do you think about this testing a private method?

File: spec/policies/person_policy_spec.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
+require 'rails_helper'
+
+RSpec.describe PersonPolicy do
+  describe '#admin?' do
+    subject { described_class.new(person, 'any object').admin? }
+    context 'the person is not signed in' do
+      let(:person) { nil }
+      it 'returns false' do
```
Comment (graygilmore) [inline spec/policies/person_policy_spec.rb]: Could rewrite as `it { is_expected.to eq false }` if you want to shorten these.

**Review by graygilmore — CHANGES_REQUESTED** [review]
Holy moly this is such a great improvement. Awesome job.

File: spec/controllers/admin_controller_spec.rb (hunk 2 above)
Comment (jacquesporveau) [reply spec/controllers/admin_controller_spec.rb]: I've changed how this test was to implemented. The test now creates a mock controller that inherits from the admin controller. Then the tests test that controller.

Comment (graygilmore) [reply spec/controllers/admin_controller_spec.rb]: I think I'd still like Brendan's opinion. Testing a private method is counter to Stembolt convention.

Comment (graygilmore) [reply spec/controllers/admin_controller_spec.rb]: After talking to Brendan about this we'd like to make use of RSpec's shared examples. What we really care about here is that the controllers that inherit from `AdminController` contain that functionality.

We don't want to have to write the same specs out every time for each controller that inherits from `AdminController`, though. Fortunately shared examples let's us define a series of assertions for the admin controller and simply include them in each controller that inherits from it.

```ruby
RSpec.describe SomeController do
  it_behaves_like "an admin controller"
end

RSpec.describe SomeOtherController do
  it_behaves_like "an admin controller"
end
```

**Comment by jacquesporveau** [comment]
jenkins test this please

**Comment by jacquesporveau** [comment]
jenkins test this please

**Pull request closed without merging**